src/azure-cli-core/azure/cli/core/style.py

```diff
@@ -46,29 +51,129 @@ class Style(str, Enum):
     Style.WARNING: Fore.LIGHTYELLOW_EX,
 }
 
+# Theme to be used on a light-themed terminal
+THEME_LIGHT = {
+    Style.PRIMARY: Fore.RESET,
+    Style.SECONDARY: Fore.LIGHTBLACK_EX,
+    Style.IMPORTANT: Fore.MAGENTA,
+    Style.ACTION: Fore.BLUE,
+    Style.HYPERLINK: Fore.CYAN,
+    Style.ERROR: Fore.RED,
+    Style.SUCCESS: Fore.GREEN,
+    Style.WARNING: Fore.YELLOW,
+}
+
+# Blue and bright blue is not visible under the default theme of powershell.exe
+POWERSHELL_COLOR_REPLACEMENT = {
+    Fore.BLUE: Fore.RESET,
+    Fore.LIGHTBLUE_EX: Fore.RESET
+}
+
 
-def print_styled_text(styled, file=sys.stderr):
-    formatted = format_styled_text(styled)
-    print(formatted, file=file)
+def print_styled_text(*styled_text_objects, file=None, **kwargs):
+    """
+    Print styled text. This function wraps the built-in function `print`, additional arguments can be sent
+    via keyword arguments.
+
+    :param styled_text_objects: The input text objects. See format_styled_text for formats of each object.
+    :param file: The file to print the styled text. The default target is sys.stderr.
+    """
+    formatted_list = [format_styled_text(obj) for obj in styled_text_objects]
+    # Always fetch the latest sys.stderr in case it has been wrapped by colorama.
+    print(*formatted_list, file=file or sys.stderr, **kwargs)
+
+
+def format_styled_text(styled_text, theme=None):
+    """Format styled text. Dark theme used by default. Available themes are 'dark', 'light', 'none'.
+
+    To change theme for all invocations of this function, set `format_styled_text.theme`.
+    To change theme for one invocation, set parameter `theme`.
+
+    :param styled_text: Can be in these formats:
+        - text
+        - (style, text)
+        - [(style, text), ...]
+    :param theme: The theme used to format text. Can be theme name or dict.
+    """
+    if theme is None:
+        theme = getattr(format_styled_text, "theme", THEME_DARK)
+
+    # Convert str to the theme dict
+    if isinstance(theme, str):
+        if theme.lower() == 'none':
+            theme = THEME_NONE
+        elif theme.lower() == 'dark':
+            theme = THEME_DARK
+        elif theme.lower() == 'light':
+            theme = THEME_LIGHT
+        else:
+            from azure.cli.core.azclierror import CLIInternalError
+            raise CLIInternalError("Invalid theme. Supported themes: none, dark, light")
```

Comment:
May I ask why do we need to support the case that `theme` is string?

By the way, could we consider setting a theme configuration of dict or enum for this mapping relationship instead of `if-else` logic? Such as:
```
THEME = {
    'none': THEME_NONE,
    'dark': THEME_DARK,
    'light': THEME_LIGHT
}
```
I personally think it might be more readable~

Reply:
> May I ask why do we need to support the case that `theme` is string?

Because everything passed to CLI via the shell will be treated as `str`.

Moreover, the `value`s of an `Enum` are usually basic types like `str` or `int`. Using `dict` as the value, like

```py
class Theme(Enum):
    DARK = THEME_DARK
    LIGHT = THEME_LIGHT
    NONE = THEME_NONE
```

will result in:
```
> az demo style -h

Arguments
    --theme            : The theme to format styled text.  Allowed values: {<Style.PRIMARY:
                         'primary'>: '\x1b[39m', <Style.SECONDARY: 'secondary'>: '\x1b[90m',
                         <Style.IMPORTANT: 'important'>: '\x1b[35m', <Style.ACTION: 'action'>:
                         '\x1b[34m', <Style.HYPERLINK: 'hyperlink'>: '\x1b[36m', <Style.ERROR:
                         'error'>: '\x1b[31m', <Style.SUCCESS: 'success'>: '\x1b[32m',
                         <Style.WARNING: 'warning'>: '\x1b[33m'}, {<Style.PRIMARY: 'primary'>:
                         '\x1b[39m', <Style.SECONDARY: 'secondary'>: '\x1b[90m', <Style.IMPORTANT:
                         'important'>: '\x1b[95m', <Style.ACTION: 'action'>: '\x1b[94m',
                         <Style.HYPERLINK: 'hyperlink'>: '\x1b[96m', <Style.ERROR: 'error'>:
                         '\x1b[91m', <Style.SUCCESS: 'success'>: '\x1b[92m', <Style.WARNING:
                         'warning'>: '\x1b[93m'}, {}.  Default: dark.
```

But the idea of using `Enum` for `Theme` is good. Added as requested:

https://github.com/Azure/azure-cli/blob/6240eafbd6829460b1a1b4a9e4a156d8675f954d/src/azure-cli-core/azure/cli/core/style.py#L67-L77

Reply:
OK